### PR TITLE
ddl2cpp command line argument for custom types

### PR DESCRIPTION
**Pull request by CJCombrink**
- Updated the ddl2cpp script to allow custom/extended types through external csv file
- Had to re-order the script to allow the command line to be parsed before setting up the parser
- Updated README

### CODE REVIEW

**Comment by CJCombrink** [comment]
I am planning to add a test yes.
I unfortunately could not get to adding one before I had to go to work today. 
We also have a public holiday tomorrow so might only attempt to add a test on Monday

**Comment by rbock** [comment]
No worries, I did not mean to rush you :-)

**Comment by CJCombrink** [comment]
@rbock Do you have some suggestion for a test?
Did you have a separate test script in mind or do you suggest a type of test that gets included with the existing `--test` command line argument?

File: tests/scripts/ddl2cpp_sample_good_custom_type.sql [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+CREATE TABLE tab_foo
```
Comment (rbock) [inline tests/scripts/ddl2cpp_sample_good_custom_type.sql:27]: Please also add a few "standard" types in addition to all the custom types, just to make sure that they are still understood when custom types are added.

**Review by rbock — CHANGES_REQUESTED** [review]
Thanks for the creating the test! I have one comment, but otherwise, this looks very good to me.

File: tests/scripts/ddl2cpp_sample_good_custom_type.sql (hunk 1 above)
Comment (CJCombrink) [reply tests/scripts/ddl2cpp_sample_good_custom_type.sql:27]: +1
Please check latest commit

Comment (rbock) [reply tests/scripts/ddl2cpp_sample_good_custom_type.sql:27]: Thanks for the update!

**Review by rbock — APPROVED** [review]

**Comment by rbock** [comment]
Looks good to me. If you're also happy at this point, I think this is ready to be merged.

**Comment by CJCombrink** [comment]
Thanks for the library and feedback.
I am happy, it can be merged.

**Comment by rbock** [comment]
Done :-)

**Pull request merged**